probot/server/tools/update-db.js

```diff
@@ -1,89 +1,89 @@
 
+const { PRtest, INFOtest } = require('../../test/utils/testmodels');
+const PR = ( process.env.TEST_ENV ? PRtest : require('../models').PR );
+const INFO = ( process.env.TEST_ENV ? INFOtest : require('../models').INFO );
+// const { getFilenames } = require('../../../lib/get-prs');
+
 // added to prevent deprecation warning when findOneAndUpdate is used
 mongoose.set('useFindAndModify', false);
-
-// connect to mongo db
-const mongoUri = config.mongo.host;
-const db = mongoose.connect(
-  mongoUri,
-  { useNewUrlParser: true }
-);
-
-const { PR, INFO } = require('../models');
-const { getPRs, getUserInput, getFilenames } = require('../../../lib/get-prs');
-const { rateLimiter } = require('../../../lib/utils');
-
-const lastUpdate = new Date();
-
-db.then(async () => {
-  const oldPRs = await PR.find({}).then(data => data);
-  // Need to add logic to stop further processing if there oldPRs does
-  // not contain any data
-  const oldIndices = oldPRs.reduce((obj, { _id }, index) => {
-    obj[_id] = index;
-    return obj;
-  }, {});
-
-  const { totalPRs, firstPR, lastPR } = await getUserInput('all');
-  const prPropsToGet = ['number', 'user', 'title', 'updated_at'];
-  const { openPRs } = await getPRs(totalPRs, firstPR, lastPR, prPropsToGet);
-  let count = 0;
-  const newIndices = {};
-  for (let i = 0; i < openPRs.length; i++) {
-    const {
+// const updateDb = async(
+  /* {
+  github,
+  issue,
+  payload: {
+    action,
+    pull_request: {
       number,
       updated_at: updatedAt,
       title,
       user: { login: username }
-    } = openPRs[i];
-
-    newIndices[number] = i;
-    let oldPrData = oldPRs[oldIndices[number]];
-    const oldUpdatedAt = oldPrData ? oldPrData.updatedAt : null;
-    if (!oldIndices.hasOwnProperty(number)) {
-      // insert a new pr
-      const filenames = await getFilenames(number);
-      await PR.create({ _id: number, updatedAt, title, username, filenames });
-      console.log('added PR# ' + number);
-    } else if (updatedAt > oldUpdatedAt) {
-      // update an existing pr
-      const filenames = await getFilenames(number);
-      count++;
-      await PR.findOneAndUpdate(
-        { _id: number },
-        { updatedAt, title, username, filenames }
-      );
-      console.log('updated PR #' + number);
-    }
-    if (count > 4000) {
-      await rateLimiter(2350);
     }
   }
-  for (let j = 0; j < oldPRs.length; j++) {
-    const { _id: number } = oldPRs[j];
-    if (!newIndices.hasOwnProperty(number)) {
-      // delete pr because it is no longer open
-      await PR.deleteOne({ _id: number });
-      console.log('deleted PR #' + number);
-    }
+}*/
+const updateDb = async(context) => {
+  const payload = context.payload;
+  const action = payload.action;
+  const pullRequest = payload.pull_request;
+  const number = pullRequest.number;
+  const updatedAt = pullRequest.updated_at;
+  const title = pullRequest.title;
+  const username = pullRequest.user.login;
```

Comment:
All of these could maybe be destructured in arguments, but I wasn't sure how.

Reply:
```
const { number, updatedAt, title, username } = pullRequest;
```

Reply:
Keep in mind that from an actual pull request it comes in as updated_at and the username comes in differently too.

Reply:
That's the fancy part I don't know how to do, so I'm leaving as-is for now.